zebra/zebra_rib.c

```diff
@@ -148,6 +149,58 @@ _rnode_zlog(const char *_func, vrf_id_t vrf_id, struct route_node *rn,
 	zlog(priority, "%s: (%u:%u):%s: %s", _func, vrf_id, table, buf, msgbuf);
 }
 
+static char *_dump_re_status(const struct route_entry *re, char *buf,
+			     size_t len)
+{
+	if (re->status == 0) {
+		snprintfrr(buf, len, "None ");
+		return buf;
+	}
+
+	snprintfrr(
+		buf, len, "%s%s%s%s%s%s%s",
+		CHECK_FLAG(re->status, ROUTE_ENTRY_REMOVED) ? "Removed " : "",
+		CHECK_FLAG(re->status, ROUTE_ENTRY_CHANGED) ? "Changed " : "",
+		CHECK_FLAG(re->status, ROUTE_ENTRY_LABELS_CHANGED)
+			? "Label Changed "
+			: "",
+		CHECK_FLAG(re->status, ROUTE_ENTRY_QUEUED) ? "Queued " : "",
+		CHECK_FLAG(re->status, ROUTE_ENTRY_INSTALLED) ? "Installed "
+							      : "",
+		CHECK_FLAG(re->status, ROUTE_ENTRY_FAILED) ? "Failed " : "",
+		CHECK_FLAG(re->status, ROUTE_ENTRY_USE_FIB_NHG) ? "Fib NHG "
+								: "");
+	return buf;
+}
+
+static char *_dump_re_flags(const struct route_entry *re, char *buf, size_t len)
+{
+	if (re->flags == 0) {
+		snprintfrr(buf, len, "None ");
+		return buf;
+	}
+
+	snprintfrr(
+		buf, len, "%s%s%s%s%s%s%s%s%s%s",
+		CHECK_FLAG(re->flags, ZEBRA_FLAG_ALLOW_RECURSION) ? "Recursion "
```

Comment:
these flags come from `lib/zclient.h` - could this formatter be in zclient.c instead, so it could be used from any of the modules that use those flags? and, for maintainability, it would be good to add a comment in `zclient.h` explaining that new flags should be added to the outputter/formatter.

Reply:
fixed